ttnn/cpp/pybind11/tensor.hpp

```diff
@@ -0,0 +1,21 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "pybind11/pybind_fwd.hpp"
+
+namespace py = pybind11;
+
+namespace ttnn::tensor {
+
+namespace detail{
+        void pytensor_module_types(pybind11::module & m_tensor);
+        void pytensor_module(pybind11::module & m_tensor);
+}
+
+    void py_module_types(py::module& module);
+    void py_module(py::module& module);
+
```

Comment:
@ayerofieiev-tt are you referring to the py_module (tensor.cpp) and pytensor_module (pytensor.cpp) module being separate ? It was kept separate earlier I am guessing for some particular reason (pytensor contains a lot of convert functions that are used inside the bindings, and it contains necessary documentation for Tensor and related functions)
As for separating 
@yan-zaretskiy suggested we keep the types( enums and classes) and function under separate binding calls and types precede function. https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/pybind11/__init__.cpp

Reply:
I did try to have it as only 2 functions - 1 for types and 1 for function call, but then again, it looks better to have pytensor.cpp separately as it is dedicated to the Tensor page in our TTNN documentation

Reply:
@KalaivaniMCW 2 things:
* Code indentation is wrong here.
* Not clear why detail namespace is in the header. I assume that functions declared in `details` namespace are only used in a corresponding pybind11/tensor.cpp, so declaration can be removed from .hpp
* It is weird from a caller standpoint to know the difference between functions `py_module` and `py_module_types` which are exposed in tensor.hpp. I recommend having a single function in a header to bind tensor (and whatever else it provides). In .cpp it can call to multiple functions declared/defined in .cpp, not exposed in .hpp